SCRUM-254 design: search input component

gdaegeun539 opened the pull request:
### Please check if the PR fulfills these requirements
- [x] The commit message follows our guidelines
<!-- - [ ] Tests for the changes have been added (for bug fixes / features) ~~  -->
<!-- - [ ] Docs have been added / updated (for bug fixes / features) -->


## What kind of change does this PR introduce?
> Select your one and delete others

- Implement design


# What is the current behavior?
>  You can also link to an open issue here

SCRUM-254
곡 추가, 아티스트 검색 등에서 사용하는 검색창 컴포넌트가 없는 상황입니다.


# What is the new behavior (if this is a feature change)?

- 검색 입력 컴포넌트를 추가합니다.
  - 검색 입력 컴포넌트에 필요한 아이콘 에셋을 추가합니다. 이들은 Tint 변경만으로 다크모드 대응이 가능합니다.
    - 검색 아이콘
    - x자 원형 아이콘


## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)

No breaking changes.


## ScreenShots (If needed)
<p>
  <img width="554" height="218" alt="스크린샷 2025-12-04 115343" src="https://github.com/user-attachments/assets/5c0c4a36-f0c1-41dd-9f1d-1300252662ce" />
</p>


## Other information:

- #23 이 머지되면 이 컴포넌트에 사용한 디자인 토큰을 변경하겠습니다.
- 사실 이번 PR은 Cursor+Opus 4.5가 바로 끝내버렸습니다. ㅇㅅㅇ...

gdaegeun539 committed:
design: add xcircle icon asset

gdaegeun539 committed:
design: add search icon asset

gdaegeun539 committed:
refactor: remove redundant xcircle dark icon asset

gdaegeun539 committed:
design: add search inputfield component

gdaegeun539 committed:
refactor: change search inputfield text style

gdaegeun539 committed:
refactor: remove unused variable

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/core/designsystem/component/FeelinSearchInputField.kt:
```diff
@@ -0,0 +1,131 @@
+                    contentDescription = null,
+                    modifier = Modifier.size(20.dp),
+                    colorFilter = ColorFilter.tint(iconTintColor),
+                )
```
`Icon`을 사용해서 벡터 이미지로 변경하는 게 좋을 것 같아요 !

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/core/designsystem/component/FeelinSearchInputField.kt:
```diff
@@ -0,0 +1,131 @@
+                                onClick = onClearClick,
+                            ),
+                    )
+                }
```
이 부분은 다른 `InputField`에서도 재사용할 수 있을 것 같아요 !

hyunjung-choi replied:
(+) 여기도 `Icon` 사용해주세요 ~

gdaegeun539 replied:
그러네요 Input_Field_Line에도 사용할 수 있네요. 빼 놓겠습니다!

hyunjung-choi replied:
이거 placeholder까지 같이 포함해서 말한 거였습니다 , , ! 😅

gdaegeun539 replied:
아하 그러면 묶어보겠습니다. `Input_Field_Line`이랑은 텍스트 속성이 아예 달라서 파라미터로 빼야 할 것 같긴 한데 간격 등을 공유한다는 점에서는 한번 더 컴포넌트로 해도 될 거 같아요.

gdaegeun539 replied:
다시 묶어봤는데 잘 묶였는지 모르겠네요... 확인 한번 부탁드립니다 🙏

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/core/designsystem/component/FeelinSearchInputField.kt:
```diff
@@ -0,0 +1,131 @@
+                                interactionSource = remember { MutableInteractionSource() },
+                                indication = null,
+                                onClick = onClearClick,
+                            ),
```
Recomposition 될 때 마다 계속 생성되지 않게 `remember`는 상단으로 빼는 게 좋을 것 같아요 !

gdaegeun539 replied:
조건 렌더링 상관없이 다른 프레임워크랑 같이 상태를 상단에 두는 게 맞군요! 수정해 두겠습니다

hyunjung-choi commented on app/src/main/java/com/lyrics/feelin/core/designsystem/component/FeelinSearchInputField.kt:
```diff
@@ -0,0 +1,131 @@
+    val placeholderColor = feelinColors.gray04
+    val iconTintColor = feelinColors.gray05
+
+    BasicTextField(
```
`imeAction`이랑 `lineLimits`를 추가해야할 것 같아요 ~

`ImeAction.Search`를 사용해서 키보드 오른쪽 아래에 있는 버튼을 "검색"이나 돋보기 아이콘으로 바꿔주세요 ! 그리고 `KeyboardActions`에 대해 좀 더 설정이 필요할 것 같아요 . . !

검색창은 보통 한 줄만 입력 가능하므로 ,, `SingleLine` 설정 해줘야해요 ,, 🙄

gdaegeun539 replied:
디자인만 맞췄더니 검색창의 요구사항을 전혀 생각 안한 문제가 생겼군요... 수정해놓겠습니다

hyunjung-choi reviewed (changes requested):
수고하셨습니다 !! 👍👍👍
몇 가지 수정 사항 제안했는데, 확인 부탁드려용

gdaegeun539 committed:
refactor: enhance searchinputfield component

- make InputClearButton component
- use icon instead image to render vector img
- add onSearchClick parameter
- add options for search field
  - lineLimits
  - keyboardOptions(imeAction, keyboard type)
  - onKeyboardAction
  - maxLength

gdaegeun539 committed:
refactor: update code lint

gdaegeun539 commented:
말씀주신 사항들 수정했습니다! 한번 다시 확인 부탁드려요

hyunjung-choi reviewed (approved):
주말인데 고생하셨습니다 ! 👍

gdaegeun539 committed:
refactor: make inner textfield to component

- remove unnecessary InputClearButton

gdaegeun539 committed:
refactor: restore modifier parameter at InnerTextFieldComponent

gdaegeun539 committed:
Merge branch 'develop' into feature/SCRUM-254-search-input

gdaegeun539 committed:
refactor: decouple M3 typo, color at new component

- FeelinSearchInputField
- InnerTextFieldComponent

gdaegeun539 commented:
확인해주셔서 감사합니다! 새로 컴포넌트로 묶은 부분도 큰 문제 없으면 머지하겠습니다.

hyunjung-choi commented:
오 ! 좋아요 !! 바로 머지합시당

gdaegeun539 commented:
> 오 ! 좋아요 !! 바로 머지합시당

감사합니다!
